src/Symfony/Component/Form/Extension/Validator/ValidatorTypeGuesser.php

```diff
@@ -122,7 +122,13 @@ public function guessTypeForConstraint(Constraint $constraint)
 
             case 'Symfony\Component\Validator\Constraints\File':
             case 'Symfony\Component\Validator\Constraints\Image':
-                return new TypeGuess('Symfony\Component\Form\Extension\Core\Type\FileType', [], Guess::HIGH_CONFIDENCE);
+                return new TypeGuess(
+                    'Symfony\Component\Form\Extension\Core\Type\FileType',
+                    ($constraint->mimeTypes && \is_array($constraint->mimeTypes)) ?
```

Comment:
In the `Image` constraint this can be a `string`. We need to take that into account too I guess.